waspc/src/Wasp/Generator/Node/Version.hs

```diff
@@ -13,25 +14,56 @@ import qualified System.Process as P
 import Text.Read (readMaybe)
 import qualified Text.Regex.TDFA as R
 import qualified Wasp.SemanticVersion as SV
+import Wasp.Util (indent)
 
+-- | Gets the installed node version, if any is installed, and checks that it
+-- meets Wasp's version requirement.
+--
+-- Returns a string representing the error
+-- condition if node's version could not be found or if the version does not
+-- meet the requirements.
+getAndCheckNodeVersion :: IO (Either String SV.Version)
```

Comment:
I am a bit late to the party, but one question here is: is this a right place to define this stuff? This is in Generator, and it makes sense because it was only Genreator that required node. But now, we are importing this logic in the rest of the app, in CLI, and using it there. We could say: that is because CLI now explicitly depends on this specific Generator! But maybe what would be nicer is extracting Node logic into higher level, something like src/Wasp/Node.hs .
Specifically, extracting the concept of Node version on which Wasp depends on, and also extracting the checks for it, that might make sense -> extracting it into src/Wasp/Node.hs as I mentioned.

I am not 100% sure about this, but it feels like the right thing to do. This way we say "not only generator depends on node, but whole Wasp", and that is really true, because Wasp even uses node to do deployment via `wasp deploy`. This is relatively new also, and we never captured that really in this way.

@sodic @craigmc08 let me know what you think.

Reply:
We could move it, makes sense if @craigmc08 is up for it?

Reply:
No problem, let's create new PR!